masschallenge/django-accelerator
# [AC-6542] Added eventbrite_organization_id field

**Changes Introduced in [AC-6542](https://masschallenge.atlassian.net/browse/AC-6542):**
- Display Eventbrite events based on program role

**Test:**

_checkout this branch on both Accelerate and Django-accelerator_
- Run `make migrate`
- Run the following in Django Shell

```python
from mc.models import *
user = User.objects.filter(pk=10000).first()
role = UserRole.objects.filter(name="Finalist").first()

program = Program.objects.filter(id=23).first()
program.eventbrite_organization_id = 109364363945
program.save()

program2 = Program.objects.filter(id=25).first()
program2.eventbrite_organization_id = 11111111111
program2.save()

pr = ProgramRole.objects.filter(user_role=role, program=program).first()
ProgramRoleGrant.objects.create(person=user, program_role=pr)

pr2 = ProgramRole.objects.filter(user_role=role, program=program2).first()
ProgramRoleGrant.objects.create(person=user, program_role=pr2)
```
- Update EVENTBRITE_KEY setting value to [this key on 1password](https://my.1password.com/vaults/yansoivyba2yqtf5h6dzh5gdbi/allitems/woeofqqatekgvecbnuloocltwe)
- Masquerade as  a [finalist user](http://localhost:8181/admin/simpleuser/user/10000/masquerade/)
- Visit the new events [url](http://localhost:8181/events/)

Notice that there is a dropdown with all the programs the user is a finalist in and that also have an eventbrite_organization_id. The first programs uses the only available organization_id and actually returns the existing events, the other uses a fake one and returns an appropriate message.

NB: Related PR to merge together https://github.com/masschallenge/accelerate/pull/2172
